File: app/src/main/java/ru/michanic/mymot/UI/Activities/LoadingActivity.java
package ru.michanic.mymot.UI.Activities;

import android.app.ActionBar;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import ru.michanic.mymot.Extensions.Font;
import ru.michanic.mymot.Interactors.ApiInteractor;
import ru.michanic.mymot.MyMotApplication;
import ru.michanic.mymot.Protocols.LoadingInterface;
import ru.michanic.mymot.R;

public class LoadingActivity extends UniversalActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_loading);
        getSupportActionBar().hide();

        MyMotApplication.setAppContext(getApplicationContext());

        TextView bigTitle = (TextView) findViewById(R.id.bigTitle);
        bigTitle.setTypeface(Font.progress);
        bigTitle.setText("Загрузка");

        TextView subtitle = (TextView) findViewById(R.id.subtitle);
        subtitle.setText("Синхронизация каталога");

        ApiInteractor apiInteractor = new ApiInteractor();
        apiInteractor.loadData(new LoadingInterface() {
            @Override
            public void onLoaded() {

                Intent mainActivity = new Intent(MyMotApplication.appContext, MainActivity.class);
                mainActivity.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
                startActivity(mainActivity);

            }
        });
    }

    private void loadCities() {
    }
}

File: app/src/main/java/ru/michanic/mymot/Kotlin/UI/Activities/LoadingActivity.kt
package ru.michanic.mymot.Kotlin.UI.Activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import ru.michanic.mymot.Kotlin.Extensions.Font
import ru.michanic.mymot.Kotlin.Interactors.ApiInteractor
import ru.michanic.mymot.Kotlin.MyMotApplication
import ru.michanic.mymot.R
import ru.michanic.mymot.databinding.ActivityLoadingBinding

class LoadingActivity : UniversalActivity() {

    private lateinit var binding: ActivityLoadingBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_loading)
        supportActionBar?.hide()
        MyMotApplication.setAppContext(applicationContext)
        val bigTitle = findViewById<View>(R.id.bigTitle) as TextView
        bigTitle.typeface = Font.progress
        bigTitle.text = "Загрузка"
        val subtitle = findViewById<View>(R.id.subtitle) as TextView
        subtitle.text = "Синхронизация каталога"
        loadData()
    }

    private fun loadData() {
        val apiInteractor = ApiInteractor()
        apiInteractor.loadData({
            val mainActivity = Intent(MyMotApplication.appContext, MainActivity::class.java)
            mainActivity.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK)
            startActivity(mainActivity)
        }, {
            showNoConnectionDialog { loadData() }
        })

    }
}